### ISSUE TITLE
Notifying nexus of live repair progress updates eats way too much CPU because of creating reqwest clients

### ISSUE THREAD

**Issue by faithanalog** [oxidecomputer/crucible#1316]
![image](https://github.com/oxidecomputer/crucible/assets/1389549/d25e79a2-7fa5-4c16-b96f-468e826a544c)

![image](https://github.com/oxidecomputer/crucible/assets/1389549/9f70eac0-2194-4d42-81ce-161d42c89caf)

This rack is very distressed. Some disks have gone into live-repair. That distress is being made even worse because **8% of the rack's CPU is being spent notifying nexus of live-repair status**, and most of that is being spent creating Reqwest clients. this is the same class of problem as https://github.com/oxidecomputer/omicron/issues/5717 which was fixed by https://github.com/oxidecomputer/omicron/pull/5750, but we are having it over here too.

Fixing this requires caching the **Reqwest client**, but it **does not require caching the nexus client**. This is an important distinction because caching the nexus client would cause problems when nexus moves around. But caching the *reqwest* client is fine, because it's just the thing that does HTTP requests, it doesn't care where they're going.

In short, we want to be calling `nexus_client::Client::new_with_client()` instead of `nexus_client::Client::new()` here https://github.com/oxidecomputer/crucible/blob/da333e91484652e3d4a4a73b94a11e5409820e04/upstairs/src/lib.rs#L1816-L1819

A Reqwest client is internally Arc'd, so we just need one laying around that we can clone every time we make a nexus client

@iliana tagging you to make sure I'm explaining this right

**Comment by iliana** [comment]
Yep that's right.
